globus_cli/safeio/output_formatter.py

```diff
@@ -97,43 +122,57 @@ def print_unix_response(res):
         click.get_current_context().exit(2)
 
 
-def colon_formatted_print(data, named_fields):
-    maxlen = max(len(n) for n, f in named_fields) + 1
-    for name, field in named_fields:
-        field_keyfunc = _key_to_keyfunc(field)
-        click.echo("{} {}".format((name + ":").ljust(maxlen), field_keyfunc(data)))
-
-
-def print_table(iterable, headers_and_keys, print_headers=True):
+def colon_formatted_print(data, fields):
+    maxlen = max(len(f.name) for f in fields) + 2
+    indent = " " * maxlen
+    wrapper = textwrap.TextWrapper(initial_indent=indent, subsequent_indent=indent)
+    for field in fields:
+        # str in case the result is `None`
+        value = str(field(data))
+
+        # 88 char wrap based on the same rationale that `black` and `flake8-bugbear`
+        # use 88 chars (or if there's a newline)
+        # only wrap if it's enabled and detected
+        shouldwrap = field.wrap_enabled and len(value) + maxlen > 88 or "\n" in value
```

Comment:
I worried for a half-second that I had this wrong, but `or` is higher precedence than `and`. ( https://docs.python.org/3/reference/expressions.html#operator-precedence )
I can add parens if this poses a readability concern.

Reply:
I generally prefer more parens over having to think about precedence, but nbd

Reply:
I'll add them. Looking at this again, I don't see a reason _not_ to add the parens. The fact that I felt the need to comment on it tells us everything we need to know -- it's potentially confusing.